### DISCUSSION TITLE
Database error saving new user

### DISCUSSION THREAD

**Discussion by ItsAlexousd** [supabase/supabase#7916] [Questions] [answered]
Hi 👋🏻, i'm currently working on a flutter app which uses supabase.
When the user creates his account, a trigger retrieves the data from his metadata to insert it in profile table.

This is my table :
```sql
create table if not exists public.profile (
    id uuid references auth.users on delete cascade not null primary key,
    username varchar(24) not null unique,
    birth_date date not null,
    height int2 not null,
    created_at timestamp with time zone default timezone('utc'::text, now()) not null
);
```

```sql
create or replace function handle_new_user() returns trigger as $$
    begin
        insert into public.profile(
            id,
            username,
            birth_date,
            height,
            )
        values(
            new.id,
            new.raw_user_meta_data->>'username',
            new.raw_user_meta_data->>'birthDate',
            new.raw_user_meta_data->>'height',
            );

        return new;
    end;
$$ language plpgsql security definer;

create trigger on_auth_user_created
    after insert on auth.users
    for each row
    execute function handle_new_user();
```

But the following error occurs in postgres logs :
`column "birth_date" is of type date but expression is of type text`
or
`column "height" is of type integer but expression is of type text`

I think that ->> returns a text. I've already tried to cast like this :
```sql
new.raw_user_meta_data->>'birthDate'::date,
new.raw_user_meta_data->>'height'::integer,
```

In the flutter console, i'm having this error :
`GotrueError(message: Database error saving new user, statusCode: null)`

Thanks a lot for your help !
Alexis.

**EDIT : solved with :** 
```sql
(new.raw_user_meta_data->>'birthDate')::date,
(new.raw_user_meta_data->'height')::integer,
```

**Comment by silentworks** [comment, marked as answer]
It's probably best to write the answer as its own reply and then mark it as answered.

**Comment by keepyara** [comment]
I was facing the same problem with boolean and the following worked for me. Thanks.
```
(new.raw_user_meta_data->>'isAdmin')::boolean,
```
